src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/SpecialFileProviders/AppManifestSpecialFileProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.IO;
+
+namespace Microsoft.VisualStudio.ProjectSystem.SpecialFileProviders
+{
+    [ExportSpecialFileProvider(SpecialFiles.AppManifest)]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasicOrFSharp)]
+    class AppManifestSpecialFileProvider : AbstractSpecialFileProvider
+    {
+        private readonly ProjectProperties _projectProperties;
+        private const string NoManifestValue = "NoManifest";
+        private const string DefaultManifestValue = "DefaultManifest";
+
+        [ImportingConstructor]
+        public AppManifestSpecialFileProvider(IPhysicalProjectTree projectTree,
+                                              [Import(ExportContractNames.ProjectItemProviders.SourceFiles)] IProjectItemProvider sourceItemsProvider,
+                                              [Import(AllowDefault = true)] Lazy<ICreateFileFromTemplateService> templateFileCreationService,
+                                              IFileSystem fileSystem,
+                                              ISpecialFilesManager specialFilesManager,
+                                              ProjectProperties projectProperties) 
+            : base(projectTree, sourceItemsProvider, templateFileCreationService, fileSystem, specialFilesManager)
+        {
+            _projectProperties = projectProperties;
+        }
+
+        protected override string Name => "app.manifest";
```

Comment:
It's used in this one place and this looks like a const anyway. Don't see the value of adding another const.